dubbo-demo-xml-provider error：dubbo rest metadata resolve double path error,and do not  contain path variable

* Dubbo version: 3.2.10
* Operating System version: centos8 aarch64
* Java version: jdk1.8
* Zookeeper version：3.9.1

### Steps to reproduce this issue

1. zkServer.sh start
2. nohup java -Djava.net.preferIPv4Stack=true -jar dubbo-demo-xml-consumer/target/dubbo-demo-xml-consumer-3.2.10.jar & sleep 
3. nohup java -Djava.net.preferIPv4Stack=true -jar dubbo-demo-xml-provider/target/dubbo-demo-xml-provider-3.2.10.jar

### Expected Behavior

Dubbo Application is ready.

### Actual Behavior

Dubbo Application has stopped.The reason for reporting the error is：”Caused by: java.lang.IllegalArgumentException: dubbo rest metadata resolve double path error,and do not  contain path variable  is“”

If there is an exception, please attach the exception trace:

```
[12/01/24 12:11:36:765 CST] main  INFO context.DubboConfigBeanInitializer:  [DUBBO] loading dubbo config beans ..., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:36:872 CST] main  INFO context.DubboConfigBeanInitializer:  [DUBBO] dubbo config beans are loaded., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:316 CST] main  INFO imps.CuratorFrameworkImpl: Starting
[12/01/24 12:11:37:331 CST] main  INFO zookeeper.ZooKeeper: Client environment:zookeeper.version=3.7.0-e3704b390a6697bfdf4b0bef79e3da7a4f6bac4b, built on 2021-03-17 09:46 UTC
[12/01/24 12:11:37:331 CST] main  INFO zookeeper.ZooKeeper: Client environment:host.name=localhost
[12/01/24 12:11:37:331 CST] main  INFO zookeeper.ZooKeeper: Client environment:java.version=1.8.0_191
[12/01/24 12:11:37:331 CST] main  INFO zookeeper.ZooKeeper: Client environment:java.vendor=AdoptOpenJdk
[12/01/24 12:11:37:331 CST] main  INFO zookeeper.ZooKeeper: Client environment:java.home=/home/spack/opt/spack/linux-centos8-aarch64/gcc-8.5.0/openjdk-1.8.0_191-b12-2pvowqoqg3nu7kwaohs6rux7tpwftzwy/jre
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:java.class.path=dubbo-demo-xml-provider/target/dubbo-demo-xml-provider-3.2.10.jar
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:java.library.path=/home/spack/opt/spack/linux-centos8-aarch64/gcc-8.5.0/maven-3.6.3-lucou5y5uw2utaaa4dcc7sr5seuz7dkn/lib:/home/spack/opt/spack/linux-centos8-aarch64/gcc-8.5.0/openjdk-1.8.0_191-b12-2pvowqoqg3nu7kwaohs6rux7tpwftzwy/lib:/home/spack/opt/spack/linux-centos8-aarch64/gcc-8.5.0/zookeeper-3.9.1-zvekqexcizqa3od67vtcz2dxs6rkttb6/lib:/usr/java/packages/lib/aarch64:/lib:/usr/lib
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:java.io.tmpdir=/tmp
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:java.compiler=<NA>
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:os.name=Linux
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:os.arch=aarch64
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:os.version=4.18.0-193.el8.aarch64
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:user.name=root
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:user.home=/root
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:user.dir=/home/stage/root/spack-stage-dubbo-3.2.10-yju25lsy37lw55ups2j4dxc2iqx4zvuu/spack-src/dubbo-demo/dubbo-demo-xml
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:os.memory.free=1529MB
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:os.memory.max=27278MB
[12/01/24 12:11:37:332 CST] main  INFO zookeeper.ZooKeeper: Client environment:os.memory.total=1556MB
[12/01/24 12:11:37:341 CST] main  INFO zookeeper.ZooKeeper: Initiating client connection, connectString=127.0.0.1:2181 sessionTimeout=60000 watcher=org.apache.curator.ConnectionState@b7dd107
[12/01/24 12:11:37:350 CST] main  INFO common.X509Util: Setting -D jdk.tls.rejectClientInitiatedRenegotiation=true to disable client-initiated TLS renegotiation
[12/01/24 12:11:37:358 CST] main  INFO zookeeper.ClientCnxnSocket: jute.maxbuffer value is 1048575 Bytes
[12/01/24 12:11:37:372 CST] main  INFO zookeeper.ClientCnxn: zookeeper.request.timeout value is 0. feature enabled=false
[12/01/24 12:11:37:381 CST] main  INFO imps.CuratorFrameworkImpl: Default schema
[12/01/24 12:11:37:385 CST] main-SendThread(127.0.0.1:2181)  INFO zookeeper.ClientCnxn: Opening socket connection to server localhost/127.0.0.1:2181.
[12/01/24 12:11:37:385 CST] main-SendThread(127.0.0.1:2181)  INFO zookeeper.ClientCnxn: SASL config status: Will not attempt to authenticate using SASL (unknown error)
[12/01/24 12:11:37:390 CST] main-SendThread(127.0.0.1:2181)  INFO zookeeper.ClientCnxn: Socket connection established, initiating session, client: /127.0.0.1:53800, server: localhost/127.0.0.1:2181
[12/01/24 12:11:37:410 CST] main-SendThread(127.0.0.1:2181)  INFO zookeeper.ClientCnxn: Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000a1b9990008, negotiated timeout = 40000
[12/01/24 12:11:37:421 CST] main-EventThread  INFO state.ConnectionStateManager: State change: CONNECTED
[12/01/24 12:11:37:422 CST] main  INFO zookeeper.ZookeeperTransporter:  [DUBBO] No valid zookeeper client found from cache, therefore create a new client for url. zookeeper://127.0.0.1:2181/org.apache.dubbo.config.ConfigCenterConfig?check=true&config-file=dubbo.properties&group=dubbo&include.spring.env=false&namespace=dubbo&timeout=30000, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:426 CST] Curator-ConnectionStateManager-0  INFO curator5.Curator5ZookeeperClient:  [DUBBO] Curator zookeeper client instance initiated successfully, session id is 10000a1b9990008, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:433 CST] main-EventThread  INFO imps.EnsembleTracker: New config event received: {}
[12/01/24 12:11:37:433 CST] main-EventThread  INFO imps.EnsembleTracker: New config event received: {}
[12/01/24 12:11:37:446 CST] main  INFO config.ConfigurationUtils:  [DUBBO] Config center was specified, but no config item found., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:446 CST] main  INFO config.ConfigurationUtils:  [DUBBO] Config center was specified, but no config item found., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:484 CST] main  INFO context.ConfigManager:  [DUBBO] The current configurations or effective configurations are as follows:, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:485 CST] main  INFO context.ConfigManager:  [DUBBO] <dubbo:application enableFileCache="true" qosPort="22222" executorManagementMode="isolation" parameters="{}" name="demo-provider" qosAcceptForeignIp="true" qosEnable="true" protocol="dubbo" />, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:488 CST] main  INFO context.ConfigManager:  [DUBBO] <dubbo:protocol preferSerialization="fastjson2,hessian2" port="-1" name="rest" />, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:488 CST] main  INFO context.ConfigManager:  [DUBBO] <dubbo:protocol preferSerialization="fastjson2,hessian2" port="-1" name="dubbo" />, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:489 CST] main  INFO context.ConfigManager:  [DUBBO] <dubbo:protocol preferSerialization="fastjson2,hessian2" port="-1" name="tri" />, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:489 CST] main  INFO context.ConfigManager:  [DUBBO] <dubbo:registry port="2181" parameters="{registry-type=service}" id="registry1" address="zookeeper://127.0.0.1:2181?registry-type=service" protocol="zookeeper" />, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:489 CST] main  INFO context.ConfigManager:  [DUBBO] <dubbo:ssl />, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:523 CST] main  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.0] has been initialized!, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:529 CST] main  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.1] has been initialized!, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:789 CST] main  INFO zookeeper.ZookeeperTransporter:  [DUBBO] find valid zookeeper client from the cache for address: zookeeper://127.0.0.1:2181/org.apache.dubbo.metadata.report.MetadataReport?application=demo-provider&file-cache=true&port=2181&protocol=zookeeper, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:790 CST] main  INFO deploy.DefaultApplicationDeployer:  [DUBBO] Dubbo Application[1.1](demo-provider) has been initialized!, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:956 CST] main  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.1] is starting., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:957 CST] main  INFO deploy.DefaultApplicationDeployer:  [DUBBO] Dubbo Application[1.1](demo-provider) is starting., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:37:979 CST] main  INFO manager.DefaultExecutorRepository:  [DUBBO] Cannot get config `export-thread-num` from module config, using default: 10, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:001 CST] main  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.0] is starting., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:001 CST] main  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.0] has started., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:002 CST] main  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.1] has started., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:002 CST] main  INFO deploy.DefaultMetricsServiceExporter:  [DUBBO] The MetricsConfig not exist, will not export metrics service., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:009 CST] main  INFO builders.InternalServiceConfigBuilder:  [DUBBO] org.apache.dubbo.metadata.MetadataServiceService Port hasn't been set will use default protocol defined in protocols., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:015 CST] main  INFO builders.InternalServiceConfigBuilder:  [DUBBO] Using dubbo protocol to export org.apache.dubbo.metadata.MetadataService service on port -1, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:059 CST] main  WARN config.ServiceConfig:  [DUBBO] Use random available port(20880) for protocol dubbo, dubbo version: 3.2.10, current host: 172.19.123.30, error code: 5-8. This may be caused by , go to https://dubbo.apache.org/faq/5/8 to find instructions. 
[12/01/24 12:11:38:542 CST] main  INFO server.Server:  [DUBBO] qos-server bind localhost:22222, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:544 CST] main  INFO config.ServiceConfig:  [DUBBO] Export dubbo service org.apache.dubbo.metadata.MetadataService to local registry url : injvm://127.0.0.1/org.apache.dubbo.metadata.MetadataService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=0&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&getAndListenInstanceMetadata.1.callback=true&getAndListenInstanceMetadata.return=true&getAndListenInstanceMetadata.sent=true&group=demo-provider&interface=org.apache.dubbo.metadata.MetadataService&methods=exportInstanceMetadata,getAndListenInstanceMetadata,getExportedServiceURLs,getExportedURLs,getInstanceMetadataChangedListenerMap,getMetadataInfo,getMetadataInfos,getMetadataURL,getServiceDefinition,getSubscribedURLs,isMetadataService,serviceName,toSortedStrings,version&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&register=false&release=3.2.10&revision=3.2.10&service.filter=-default&side=provider&timestamp=1705032698043&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:545 CST] main  INFO config.ServiceConfig:  [DUBBO] Export dubbo service org.apache.dubbo.metadata.MetadataService to url dubbo://172.19.123.30:20880/org.apache.dubbo.metadata.MetadataService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=0&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&getAndListenInstanceMetadata.1.callback=true&getAndListenInstanceMetadata.return=true&getAndListenInstanceMetadata.sent=true&group=demo-provider&interface=org.apache.dubbo.metadata.MetadataService&methods=exportInstanceMetadata,getAndListenInstanceMetadata,getExportedServiceURLs,getExportedURLs,getInstanceMetadataChangedListenerMap,getMetadataInfo,getMetadataInfos,getMetadataURL,getServiceDefinition,getSubscribedURLs,isMetadataService,serviceName,toSortedStrings,version&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&register=false&release=3.2.10&revision=3.2.10&service.filter=-default&side=provider&timestamp=1705032698043&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:639 CST] main  INFO transport.AbstractServer:  [DUBBO] Start NettyServer bind /0.0.0.0:20880, export /172.19.123.30:20880, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:651 CST] main  INFO metadata.ConfigurableMetadataServiceExporter:  [DUBBO] The MetadataService exports urls : [dubbo://172.19.123.30:20880/org.apache.dubbo.metadata.MetadataService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=0&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&getAndListenInstanceMetadata.1.callback=true&getAndListenInstanceMetadata.return=true&getAndListenInstanceMetadata.sent=true&group=demo-provider&interface=org.apache.dubbo.metadata.MetadataService&methods=exportInstanceMetadata,getAndListenInstanceMetadata,getExportedServiceURLs,getExportedURLs,getInstanceMetadataChangedListenerMap,getMetadataInfo,getMetadataInfos,getMetadataURL,getServiceDefinition,getSubscribedURLs,isMetadataService,serviceName,toSortedStrings,version&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&register=false&release=3.2.10&revision=3.2.10&service.filter=-default&side=provider&timestamp=1705032698043&version=1.0.0], dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:652 CST] main  INFO metadata.ServiceInstanceMetadataUtils:  [DUBBO] Start registering instance address to registry., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:662 CST] main  INFO deploy.DefaultApplicationDeployer:  [DUBBO] Dubbo Application[1.1](demo-provider) is ready., dubbo version: 3.2.10, current host: 172.19.123.30
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)
Caused by: java.io.IOException: Bad file descriptor
	at java.io.FileInputStream.readBytes(Native Method)
	at java.io.FileInputStream.read(FileInputStream.java:255)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:265)
	at org.apache.dubbo.demo.provider.Application.main(Application.java:25)
	... 8 more
[12/01/24 12:11:38:668 CST] DubboShutdownHook  INFO config.DubboShutdownHook:  [DUBBO] Run shutdown hook now., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:38:668 CST] DubboShutdownHook  INFO config.DubboShutdownHook:  [DUBBO] Waiting for modules(Dubbo Application[1.1](demo-provider)) managed by Spring to be shutdown., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:234 CST] Dubbo-demo-provider-service-export-thread-2  WARN config.ServiceConfig:  [DUBBO] Use random available port(80) for protocol rest, dubbo version: 3.2.10, current host: 172.19.123.30, error code: 5-8. This may be caused by , go to https://dubbo.apache.org/faq/5/8 to find instructions. 
[12/01/24 12:11:43:252 CST] Dubbo-demo-provider-service-export-thread-3  WARN config.ServiceConfig:  [DUBBO] Use random available port(50051) for protocol tri, dubbo version: 3.2.10, current host: 172.19.123.30, error code: 5-8. This may be caused by , go to https://dubbo.apache.org/faq/5/8 to find instructions. 
[12/01/24 12:11:43:294 CST] Dubbo-demo-provider-service-export-thread-3  INFO config.ServiceConfig:  [DUBBO] Export dubbo service org.apache.dubbo.demo.TripleService to local registry url : injvm://127.0.0.1/org.apache.dubbo.demo.TripleService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=50051&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&interface=org.apache.dubbo.demo.TripleService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:294 CST] Dubbo-demo-provider-service-export-thread-1  INFO config.ServiceConfig:  [DUBBO] Export dubbo service org.apache.dubbo.demo.GreetingService to local registry url : injvm://127.0.0.1/org.apache.dubbo.demo.GreetingService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&group=greeting&interface=org.apache.dubbo.demo.GreetingService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:294 CST] Dubbo-demo-provider-service-export-thread-2  INFO config.ServiceConfig:  [DUBBO] Export dubbo service org.apache.dubbo.demo.RestDemoService to local registry url : injvm://127.0.0.1/org.apache.dubbo.demo.RestDemoService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=80&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&interface=org.apache.dubbo.demo.RestDemoService&methods=error,getRemoteApplicationName,hello,sayHello,testBody,testBody2,testBody5,testForm1,testForm2&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:294 CST] Dubbo-demo-provider-service-export-thread-4  INFO config.ServiceConfig:  [DUBBO] Export dubbo service org.apache.dubbo.demo.DemoService to local registry url : injvm://127.0.0.1/org.apache.dubbo.demo.DemoService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&interface=org.apache.dubbo.demo.DemoService&methods=sayHello,sayHelloAsync&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&side=provider&timeout=3000&timestamp=1705032703198, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:295 CST] Dubbo-demo-provider-service-export-thread-3  INFO config.ServiceConfig:  [DUBBO] Register dubbo service org.apache.dubbo.demo.TripleService url tri://172.19.123.30:50051/org.apache.dubbo.demo.TripleService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=50051&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=org.apache.dubbo.demo.TripleService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0 to registry 127.0.0.1:2181, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:295 CST] Dubbo-demo-provider-service-export-thread-1  INFO config.ServiceConfig:  [DUBBO] Register dubbo service org.apache.dubbo.demo.GreetingService url dubbo://172.19.123.30:20880/org.apache.dubbo.demo.GreetingService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&group=greeting&interface=org.apache.dubbo.demo.GreetingService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0 to registry 127.0.0.1:2181, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:295 CST] Dubbo-demo-provider-service-export-thread-2  INFO config.ServiceConfig:  [DUBBO] Register dubbo service org.apache.dubbo.demo.RestDemoService url rest://172.19.123.30:80/org.apache.dubbo.demo.RestDemoService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=80&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=org.apache.dubbo.demo.RestDemoService&methods=error,getRemoteApplicationName,hello,sayHello,testBody,testBody2,testBody5,testForm1,testForm2&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0 to registry 127.0.0.1:2181, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:295 CST] Dubbo-demo-provider-service-export-thread-4  INFO config.ServiceConfig:  [DUBBO] Register dubbo service org.apache.dubbo.demo.DemoService url dubbo://172.19.123.30:20880/org.apache.dubbo.demo.DemoService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=org.apache.dubbo.demo.DemoService&methods=sayHello,sayHelloAsync&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=3000&timestamp=1705032703198 to registry 127.0.0.1:2181, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:318 CST] Dubbo-demo-provider-service-export-thread-3  INFO tri.TripleProtocol:  [DUBBO] Add fallback triple invoker[tri://172.19.123.30:50051/org.apache.dubbo.demo.TripleService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=50051&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=org.apache.dubbo.demo.TripleService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0] to path[org.apache.dubbo.demo.TripleService] with invoker[tri://172.19.123.30:50051/org.apache.dubbo.demo.TripleService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=50051&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=org.apache.dubbo.demo.TripleService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0], dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:510 CST] Dubbo-demo-provider-service-export-thread-3  INFO transport.AbstractServer:  [DUBBO] Start NettyPortUnificationServer bind /0.0.0.0:50051, export /0.0.0.0:50051, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:523 CST] Dubbo-demo-provider-service-export-thread-2 ERROR config.ServiceConfig:  [DUBBO] Failed to (async)export service config: org.apache.dubbo.demo.RestDemoService, dubbo version: 3.2.10, current host: 172.19.123.30, error code: 5-9. This may be caused by configuration server disconnected, go to https://dubbo.apache.org/faq/5/9 to find instructions. 
java.lang.RuntimeException: java.lang.IllegalArgumentException: dubbo rest metadata resolve double path error,and do not  contain path variable  is: PathMatcher{path='/demoService/testBody3', version='1.0.0', group='null', port=null, hasPathVariable=false, contextPath='null', httpMethod='POST'}, rest method metadata is: RestMethodMetadata{method=MethodDefinition [name=testBody2, parameterTypes=[po.TestPO], returnType=po.TestPO], request=RequestMetadata{method='POST', path='/demoService/testBody3', params={}, headers={}, consumes=[text/plain], produces=[]}, urlIndex=null, bodyIndex=null, headerMapIndex=null, bodyType='null', indexToName={0=[b]}, formParams=null, indexToEncoded=null, argInfos=[ArgInfo{index=0, annotationNameAttribute='b', paramAnnotationType=interface org.apache.dubbo.metadata.rest.tag.BodyTag, paramType=class po.TestPO, paramName='b', urlSplitIndex=0, defaultValue=null, formContentType=false}], reflectMethod=public abstract po.TestPO org.apache.dubbo.demo.RestDemoService.testBody2(po.TestPO), codeStyle=class org.apache.dubbo.metadata.rest.jaxrs.JAXRSServiceRestMetadataResolver}
	at org.apache.dubbo.registry.integration.ExporterFactory.lambda$createExporter$0(ExporterFactory.java:33)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.dubbo.registry.integration.ExporterFactory.createExporter(ExporterFactory.java:29)
	at org.apache.dubbo.registry.integration.RegistryProtocol.doLocalExport(RegistryProtocol.java:356)
	at org.apache.dubbo.registry.integration.RegistryProtocol.export(RegistryProtocol.java:291)
	at org.apache.dubbo.qos.protocol.QosProtocolWrapper.export(QosProtocolWrapper.java:79)
	at org.apache.dubbo.rpc.protocol.ProtocolListenerWrapper.export(ProtocolListenerWrapper.java:66)
	at org.apache.dubbo.rpc.protocol.ProtocolSecurityWrapper.export(ProtocolSecurityWrapper.java:84)
	at org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper.export(ProtocolFilterWrapper.java:58)
	at org.apache.dubbo.rpc.protocol.ProtocolSerializationWrapper.export(ProtocolSerializationWrapper.java:50)
	at org.apache.dubbo.rpc.protocol.InvokerCountWrapper.export(InvokerCountWrapper.java:42)
	at org.apache.dubbo.rpc.Protocol$Adaptive.export(Protocol$Adaptive.java)
	at org.apache.dubbo.config.ServiceConfig.doExportUrl(ServiceConfig.java:944)
	at org.apache.dubbo.config.ServiceConfig.exportRemote(ServiceConfig.java:914)
	at org.apache.dubbo.config.ServiceConfig.exportUrl(ServiceConfig.java:854)
	at org.apache.dubbo.config.ServiceConfig.doExportUrlsFor1Protocol(ServiceConfig.java:598)
	at org.apache.dubbo.config.ServiceConfig.doExportUrls(ServiceConfig.java:579)
	at org.apache.dubbo.config.ServiceConfig.doExport(ServiceConfig.java:540)
	at org.apache.dubbo.config.ServiceConfig.lambda$doDelayExport$0(ServiceConfig.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: dubbo rest metadata resolve double path error,and do not  contain path variable  is: PathMatcher{path='/demoService/testBody3', version='1.0.0', group='null', port=null, hasPathVariable=false, contextPath='null', httpMethod='POST'}, rest method metadata is: RestMethodMetadata{method=MethodDefinition [name=testBody2, parameterTypes=[po.TestPO], returnType=po.TestPO], request=RequestMetadata{method='POST', path='/demoService/testBody3', params={}, headers={}, consumes=[text/plain], produces=[]}, urlIndex=null, bodyIndex=null, headerMapIndex=null, bodyType='null', indexToName={0=[b]}, formParams=null, indexToEncoded=null, argInfos=[ArgInfo{index=0, annotationNameAttribute='b', paramAnnotationType=interface org.apache.dubbo.metadata.rest.tag.BodyTag, paramType=class po.TestPO, paramName='b', urlSplitIndex=0, defaultValue=null, formContentType=false}], reflectMethod=public abstract po.TestPO org.apache.dubbo.demo.RestDemoService.testBody2(po.TestPO), codeStyle=class org.apache.dubbo.metadata.rest.jaxrs.JAXRSServiceRestMetadataResolver}
	at org.apache.dubbo.metadata.rest.ServiceRestMetadata.doublePathCheck(ServiceRestMetadata.java:152)
	at org.apache.dubbo.metadata.rest.ServiceRestMetadata.addPathToServiceMap(ServiceRestMetadata.java:136)
	at org.apache.dubbo.metadata.rest.ServiceRestMetadata.addRestMethodMetadata(ServiceRestMetadata.java:118)
	at org.apache.dubbo.metadata.rest.AbstractServiceRestMetadataResolver.processRestMethodMetadata(AbstractServiceRestMetadataResolver.java:341)
	at org.apache.dubbo.metadata.rest.AbstractServiceRestMetadataResolver.processAllRestMethodMetadata(AbstractServiceRestMetadataResolver.java:170)
	at org.apache.dubbo.metadata.rest.AbstractServiceRestMetadataResolver.resolve(AbstractServiceRestMetadataResolver.java:127)
	at org.apache.dubbo.rpc.protocol.rest.annotation.metadata.MetadataResolver.resolveProviderServiceMetadata(MetadataResolver.java:69)
	at org.apache.dubbo.rpc.protocol.rest.RestProtocol.export(RestProtocol.java:88)
	at org.apache.dubbo.qos.protocol.QosProtocolWrapper.export(QosProtocolWrapper.java:79)
	at org.apache.dubbo.rpc.protocol.ProtocolListenerWrapper.export(ProtocolListenerWrapper.java:78)
	at org.apache.dubbo.rpc.protocol.ProtocolSecurityWrapper.export(ProtocolSecurityWrapper.java:84)
	at org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper.export(ProtocolFilterWrapper.java:61)
	at org.apache.dubbo.rpc.protocol.ProtocolSerializationWrapper.export(ProtocolSerializationWrapper.java:50)
	at org.apache.dubbo.rpc.protocol.InvokerCountWrapper.export(InvokerCountWrapper.java:42)
	at org.apache.dubbo.rpc.Protocol$Adaptive.export(Protocol$Adaptive.java)
	at org.apache.dubbo.registry.integration.RegistryProtocol.lambda$doLocalExport$5(RegistryProtocol.java:356)
	at org.apache.dubbo.registry.integration.ExporterFactory.lambda$createExporter$0(ExporterFactory.java:31)
	... 25 more
[12/01/24 12:11:43:526 CST] Dubbo-demo-provider-service-export-thread-1  INFO imps.CuratorFrameworkImpl: Starting
[12/01/24 12:11:43:527 CST] Dubbo-demo-provider-service-export-thread-1  INFO zookeeper.ZooKeeper: Initiating client connection, connectString=127.0.0.1:2181 sessionTimeout=60000 watcher=org.apache.curator.ConnectionState@34252bf4
[12/01/24 12:11:43:527 CST] Dubbo-demo-provider-service-export-thread-1  INFO zookeeper.ClientCnxnSocket: jute.maxbuffer value is 1048575 Bytes
[12/01/24 12:11:43:527 CST] Dubbo-demo-provider-service-export-thread-1  INFO zookeeper.ClientCnxn: zookeeper.request.timeout value is 0. feature enabled=false
[12/01/24 12:11:43:528 CST] Dubbo-demo-provider-service-export-thread-1  INFO imps.CuratorFrameworkImpl: Default schema
[12/01/24 12:11:43:528 CST] Dubbo-demo-provider-service-export-thread-1-SendThread(127.0.0.1:2181)  INFO zookeeper.ClientCnxn: Opening socket connection to server localhost/127.0.0.1:2181.
[12/01/24 12:11:43:528 CST] Dubbo-demo-provider-service-export-thread-1-SendThread(127.0.0.1:2181)  INFO zookeeper.ClientCnxn: SASL config status: Will not attempt to authenticate using SASL (unknown error)
[12/01/24 12:11:43:529 CST] Dubbo-demo-provider-service-export-thread-1-SendThread(127.0.0.1:2181)  INFO zookeeper.ClientCnxn: Socket connection established, initiating session, client: /127.0.0.1:53804, server: localhost/127.0.0.1:2181
[12/01/24 12:11:43:543 CST] Dubbo-demo-provider-service-export-thread-1-SendThread(127.0.0.1:2181)  INFO zookeeper.ClientCnxn: Session establishment complete on server localhost/127.0.0.1:2181, session id = 0x10000a1b9990009, negotiated timeout = 40000
[12/01/24 12:11:43:543 CST] Dubbo-demo-provider-service-export-thread-1-EventThread  INFO state.ConnectionStateManager: State change: CONNECTED
[12/01/24 12:11:43:545 CST] Dubbo-demo-provider-service-export-thread-1-EventThread  INFO imps.EnsembleTracker: New config event received: {}
[12/01/24 12:11:43:545 CST] Dubbo-demo-provider-service-export-thread-1-EventThread  INFO imps.EnsembleTracker: New config event received: {}
[12/01/24 12:11:43:823 CST] Dubbo-demo-provider-service-export-thread-3  INFO migration.MigrationRuleListener:  [DUBBO] Listening for migration rules on dataId demo-provider.migration, group DUBBO_SERVICEDISCOVERY_MIGRATION, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:835 CST] Dubbo-demo-provider-service-export-thread-3  INFO config.ServiceConfig:  [DUBBO] Try to register interface application mapping for service org.apache.dubbo.demo.TripleService:1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:835 CST] Dubbo-demo-provider-service-export-thread-1  INFO config.ServiceConfig:  [DUBBO] Try to register interface application mapping for service greeting/org.apache.dubbo.demo.GreetingService:1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:835 CST] Dubbo-demo-provider-service-export-thread-4  INFO config.ServiceConfig:  [DUBBO] Try to register interface application mapping for service org.apache.dubbo.demo.DemoService, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:836 CST] DubboSaveMetadataReport-thread-1  INFO zookeeper.ZookeeperMetadataReport:  [DUBBO] store provider metadata. Identifier : MetadataIdentifier{application='demo-provider', serviceInterface='org.apache.dubbo.demo.GreetingService', version='1.0.0', group='greeting', side='provider'}; definition: FullServiceDefinition{parameters={dubbo=2.0.2, group=greeting, side=provider, anyhost=true, release=3.2.10, application=demo-provider, pid=291822, interface=org.apache.dubbo.demo.GreetingService, version=1.0.0, executor-management-mode=isolation, file-cache=true, methods=hello, deprecated=false, qos.port=22222, service-name-mapping=true, qos.enable=true, timeout=5000, bind.port=20880, generic=false, revision=3.2.10, delay=5000, bind.ip=172.19.123.30, prefer.serialization=fastjson2,hessian2, background=false, dynamic=true, qos.accept.foreign.ip=true, timestamp=1705032703198}} ServiceDefinition [canonicalName=org.apache.dubbo.demo.GreetingService, codeSource=jar:file:/home/stage/root/spack-stage-dubbo-3.2.10-yju25lsy37lw55ups2j4dxc2iqx4zvuu/spack-src/dubbo-demo/dubbo-demo-xml/dubbo-demo-xml-provider/target/dubbo-demo-xml-provider-3.2.10.jar!/BOOT-INF/lib/dubbo-demo-interface-3.2.10.jar!/, methods=[MethodDefinition [name=hello, parameterTypes=[], returnType=java.lang.String]]], dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:847 CST] Dubbo-demo-provider-service-export-thread-3  INFO config.ServiceConfig:  [DUBBO] Successfully registered interface application mapping for service org.apache.dubbo.demo.TripleService:1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:847 CST] Dubbo-demo-provider-service-export-thread-4  INFO config.ServiceConfig:  [DUBBO] Successfully registered interface application mapping for service org.apache.dubbo.demo.DemoService, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:847 CST] Dubbo-demo-provider-service-export-thread-1  INFO config.ServiceConfig:  [DUBBO] Successfully registered interface application mapping for service greeting/org.apache.dubbo.demo.GreetingService:1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:893 CST] DubboSaveMetadataReport-thread-1  INFO zookeeper.ZookeeperMetadataReport:  [DUBBO] store provider metadata. Identifier : MetadataIdentifier{application='demo-provider', serviceInterface='org.apache.dubbo.demo.TripleService', version='1.0.0', group='', side='provider'}; definition: FullServiceDefinition{parameters={dubbo=2.0.2, side=provider, anyhost=true, release=3.2.10, application=demo-provider, pid=291822, interface=org.apache.dubbo.demo.TripleService, version=1.0.0, executor-management-mode=isolation, file-cache=true, methods=hello, deprecated=false, qos.port=22222, service-name-mapping=true, qos.enable=true, timeout=5000, bind.port=50051, generic=false, revision=3.2.10, delay=5000, bind.ip=172.19.123.30, prefer.serialization=fastjson2,hessian2, background=false, dynamic=true, qos.accept.foreign.ip=true, timestamp=1705032703198}} ServiceDefinition [canonicalName=org.apache.dubbo.demo.TripleService, codeSource=jar:file:/home/stage/root/spack-stage-dubbo-3.2.10-yju25lsy37lw55ups2j4dxc2iqx4zvuu/spack-src/dubbo-demo/dubbo-demo-xml/dubbo-demo-xml-provider/target/dubbo-demo-xml-provider-3.2.10.jar!/BOOT-INF/lib/dubbo-demo-interface-3.2.10.jar!/, methods=[MethodDefinition [name=hello, parameterTypes=[], returnType=java.lang.String]]], dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:43:918 CST] DubboSaveMetadataReport-thread-1  INFO zookeeper.ZookeeperMetadataReport:  [DUBBO] store provider metadata. Identifier : MetadataIdentifier{application='demo-provider', serviceInterface='org.apache.dubbo.demo.DemoService', version='', group='', side='provider'}; definition: FullServiceDefinition{parameters={dubbo=2.0.2, side=provider, anyhost=true, release=3.2.10, application=demo-provider, pid=291822, interface=org.apache.dubbo.demo.DemoService, executor-management-mode=isolation, file-cache=true, methods=sayHello,sayHelloAsync, deprecated=false, qos.port=22222, service-name-mapping=true, qos.enable=true, timeout=3000, bind.port=20880, generic=false, revision=3.2.10, delay=5000, bind.ip=172.19.123.30, prefer.serialization=fastjson2,hessian2, background=false, dynamic=true, qos.accept.foreign.ip=true, timestamp=1705032703198}} ServiceDefinition [canonicalName=org.apache.dubbo.demo.DemoService, codeSource=jar:file:/home/stage/root/spack-stage-dubbo-3.2.10-yju25lsy37lw55ups2j4dxc2iqx4zvuu/spack-src/dubbo-demo/dubbo-demo-xml/dubbo-demo-xml-provider/target/dubbo-demo-xml-provider-3.2.10.jar!/BOOT-INF/lib/dubbo-demo-interface-3.2.10.jar!/, methods=[MethodDefinition [name=sayHello, parameterTypes=[java.lang.String], returnType=java.lang.String], MethodDefinition [name=sayHelloAsync, parameterTypes=[java.lang.String], returnType=java.util.concurrent.CompletableFuture]]], dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:44:686 CST] Dubbo-framework-shared-scheduler-thread-1  INFO metadata.MetadataInfo:  [DUBBO] metadata revision changed: null -> dbdb95476a0b190157776519f15fcfc5, app: demo-provider, services: 3, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:676 CST] DubboShutdownHook  INFO config.DubboShutdownHook:  [DUBBO] Dubbo wait for application(Dubbo Application[1.1](demo-provider)) managed by Spring to be shutdown failed, time usage: 10008ms, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:676 CST] DubboShutdownHook  INFO config.DubboShutdownHook:  [DUBBO] Dubbo shutdown hooks execute now. Dubbo Application[1.1](demo-provider), dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:677 CST] DubboShutdownHook  INFO model.FrameworkModel:  [DUBBO] Reset global default application from Dubbo Application[1.1](demo-provider) to null, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:677 CST] DubboShutdownHook  INFO deploy.DefaultApplicationDeployer:  [DUBBO] Dubbo Application[1.1](demo-provider) is stopping., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:696 CST] DubboShutdownHook  INFO dubbo.DubboProtocol:  [DUBBO] Destroying protocol [DubboProtocol] ..., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:696 CST] DubboShutdownHook  INFO dubbo.DubboProtocol:  [DUBBO] Closing dubbo server: /172.19.123.30:20880, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:697 CST] DubboShutdownHook  INFO transport.AbstractServer:  [DUBBO] Close NettyServer bind /0.0.0.0:20880, export /172.19.123.30:20880, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:807 CST] DubboShutdownHook  INFO dubbo.DubboProtocol:  [DUBBO] Unexport service: dubbo://172.19.123.30:20880/org.apache.dubbo.metadata.MetadataService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=0&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&getAndListenInstanceMetadata.1.callback=true&getAndListenInstanceMetadata.return=true&getAndListenInstanceMetadata.sent=true&group=demo-provider&interface=org.apache.dubbo.metadata.MetadataService&methods=exportInstanceMetadata,getAndListenInstanceMetadata,getExportedServiceURLs,getExportedURLs,getInstanceMetadataChangedListenerMap,getMetadataInfo,getMetadataInfos,getMetadataURL,getServiceDefinition,getSubscribedURLs,isMetadataService,serviceName,toSortedStrings,version&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&register=false&release=3.2.10&revision=3.2.10&service.filter=-default&side=provider&timestamp=1705032698043&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:807 CST] DubboShutdownHook  INFO dubbo.DubboProtocol:  [DUBBO] Unexport service: dubbo://172.19.123.30:20880/org.apache.dubbo.demo.DemoService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=org.apache.dubbo.demo.DemoService&methods=sayHello,sayHelloAsync&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=3000&timestamp=1705032703198, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:807 CST] DubboShutdownHook  INFO dubbo.DubboProtocol:  [DUBBO] Unexport service: dubbo://172.19.123.30:20880/org.apache.dubbo.demo.GreetingService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&group=greeting&interface=org.apache.dubbo.demo.GreetingService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:807 CST] DubboShutdownHook  INFO server.Server:  [DUBBO] qos-server stopped., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:827 CST] DubboShutdownHook  INFO injvm.InjvmProtocol:  [DUBBO] Unexport service: injvm://127.0.0.1/org.apache.dubbo.metadata.MetadataService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=0&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&getAndListenInstanceMetadata.1.callback=true&getAndListenInstanceMetadata.return=true&getAndListenInstanceMetadata.sent=true&group=demo-provider&interface=org.apache.dubbo.metadata.MetadataService&methods=exportInstanceMetadata,getAndListenInstanceMetadata,getExportedServiceURLs,getExportedURLs,getInstanceMetadataChangedListenerMap,getMetadataInfo,getMetadataInfos,getMetadataURL,getServiceDefinition,getSubscribedURLs,isMetadataService,serviceName,toSortedStrings,version&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&register=false&release=3.2.10&revision=3.2.10&service.filter=-default&side=provider&timestamp=1705032698043&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:827 CST] DubboShutdownHook  INFO injvm.InjvmProtocol:  [DUBBO] Unexport service: injvm://127.0.0.1/org.apache.dubbo.demo.TripleService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=50051&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&interface=org.apache.dubbo.demo.TripleService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:827 CST] DubboShutdownHook  INFO injvm.InjvmProtocol:  [DUBBO] Unexport service: injvm://127.0.0.1/org.apache.dubbo.demo.RestDemoService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=80&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&interface=org.apache.dubbo.demo.RestDemoService&methods=error,getRemoteApplicationName,hello,sayHello,testBody,testBody2,testBody5,testForm1,testForm2&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:827 CST] DubboShutdownHook  INFO injvm.InjvmProtocol:  [DUBBO] Unexport service: injvm://127.0.0.1/org.apache.dubbo.demo.DemoService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&interface=org.apache.dubbo.demo.DemoService&methods=sayHello,sayHelloAsync&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&side=provider&timeout=3000&timestamp=1705032703198, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:827 CST] DubboShutdownHook  INFO injvm.InjvmProtocol:  [DUBBO] Unexport service: injvm://127.0.0.1/org.apache.dubbo.demo.GreetingService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=20880&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&exporter.listener=injvm&file-cache=true&generic=false&group=greeting&interface=org.apache.dubbo.demo.GreetingService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:827 CST] DubboShutdownHook  INFO rest.RestProtocol:  [DUBBO] Destroying protocol [RestProtocol] ..., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:827 CST] DubboShutdownHook  INFO rest.RestProtocol:  [DUBBO] Closing rest clients, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:831 CST] DubboShutdownHook  INFO tri.TripleProtocol:  [DUBBO] Destroying protocol [TripleProtocol] ..., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:831 CST] DubboShutdownHook  INFO tri.TripleProtocol:  [DUBBO] Destroy reference: tri://172.19.123.30:50051/org.apache.dubbo.demo.TripleService?anyhost=true&application=demo-provider&background=false&bind.ip=172.19.123.30&bind.port=50051&delay=5000&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=org.apache.dubbo.demo.TripleService&methods=hello&pid=291822&prefer.serialization=fastjson2,hessian2&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&release=3.2.10&revision=3.2.10&service-name-mapping=true&side=provider&timeout=5000&timestamp=1705032703198&version=1.0.0, dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:832 CST] DubboShutdownHook  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.1] is stopping., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:833 CST] DubboShutdownHook  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.1] has stopped., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:834 CST] DubboShutdownHook  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.0] is stopping., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:834 CST] DubboShutdownHook  INFO deploy.DefaultModuleDeployer:  [DUBBO] Dubbo Module[1.1.0] has stopped., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:835 CST] DubboShutdownHook  INFO support.RegistryManager:  [DUBBO] Close all registries [zookeeper://127.0.0.1:2181/org.apache.dubbo.registry.RegistryService?REGISTRY_CLUSTER=registry1&application=demo-provider&dubbo=2.0.2&executor-management-mode=isolation&file-cache=true&interface=org.apache.dubbo.registry.RegistryService&pid=291822&qos.accept.foreign.ip=true&qos.enable=true&qos.port=22222&registry-type=service&registry.type=service&release=3.2.10], dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:841 CST] Curator-Framework-0  INFO imps.CuratorFrameworkImpl: backgroundOperationsLoop exiting
[12/01/24 12:11:48:955 CST] DubboShutdownHook  INFO zookeeper.ZooKeeper: Session: 0x10000a1b9990009 closed
[12/01/24 12:11:48:955 CST] Dubbo-demo-provider-service-export-thread-1-EventThread  INFO zookeeper.ClientCnxn: EventThread shut down for session: 0x10000a1b9990009
[12/01/24 12:11:48:957 CST] DubboShutdownHook  INFO manager.DefaultExecutorRepository:  [DUBBO] destroying application executor repository .., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:957 CST] DubboShutdownHook  INFO deploy.DefaultApplicationDeployer:  [DUBBO] Dubbo Application[1.1](demo-provider) has stopped., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:48:957 CST] Curator-Framework-0  INFO imps.CuratorFrameworkImpl: backgroundOperationsLoop exiting
[12/01/24 12:11:49:065 CST] DubboShutdownHook  INFO zookeeper.ZooKeeper: Session: 0x10000a1b9990008 closed
[12/01/24 12:11:49:065 CST] main-EventThread ERROR cache.NodeCache: 
java.lang.IllegalStateException: Expected state [STARTED] was [STOPPED]
	at org.apache.curator.shaded.com.google.common.base.Preconditions.checkState(Preconditions.java:823)
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.checkState(CuratorFrameworkImpl.java:423)
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.checkExists(CuratorFrameworkImpl.java:450)
	at org.apache.curator.framework.recipes.cache.NodeCache.reset(NodeCache.java:261)
	at org.apache.curator.framework.recipes.cache.NodeCache.access$100(NodeCache.java:60)
	at org.apache.curator.framework.recipes.cache.NodeCache$2.process(NodeCache.java:104)
	at org.apache.curator.framework.imps.NamespaceWatcher.process(NamespaceWatcher.java:77)
	at org.apache.zookeeper.ClientCnxn$EventThread.processEvent(ClientCnxn.java:580)
	at org.apache.zookeeper.ClientCnxn$EventThread.run(ClientCnxn.java:555)
[12/01/24 12:11:49:065 CST] DubboShutdownHook  INFO model.FrameworkModel:  [DUBBO] Destroying default framework model: Dubbo Framework[1], dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:49:065 CST] main-EventThread ERROR cache.NodeCache: 
java.lang.IllegalStateException: Expected state [STARTED] was [STOPPED]
	at org.apache.curator.shaded.com.google.common.base.Preconditions.checkState(Preconditions.java:823)
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.checkState(CuratorFrameworkImpl.java:423)
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.checkExists(CuratorFrameworkImpl.java:450)
	at org.apache.curator.framework.recipes.cache.NodeCache.reset(NodeCache.java:261)
	at org.apache.curator.framework.recipes.cache.NodeCache.access$100(NodeCache.java:60)
	at org.apache.curator.framework.recipes.cache.NodeCache$2.process(NodeCache.java:104)
	at org.apache.curator.framework.imps.NamespaceWatcher.process(NamespaceWatcher.java:77)
	at org.apache.zookeeper.ClientCnxn$EventThread.processEvent(ClientCnxn.java:580)
	at org.apache.zookeeper.ClientCnxn$EventThread.run(ClientCnxn.java:555)
[12/01/24 12:11:49:065 CST] DubboShutdownHook  INFO model.FrameworkModel:  [DUBBO] Dubbo Framework[1] is destroying ..., dubbo version: 3.2.10, current host: 172.19.123.30
[12/01/24 12:11:49:065 CST] main-EventThread ERROR cache.NodeCache: 
java.lang.IllegalStateException: Expected state [STARTED] was [STOPPED]
```

--- a/dubbo-demo/dubbo-demo-interface/src/main/java/org/apache/dubbo/demo/RestDemoService.java
+++ b/dubbo-demo/dubbo-demo-interface/src/main/java/org/apache/dubbo/demo/RestDemoService.java
@@ -65,7 +65,7 @@ public interface RestDemoService {
     Boolean testBody2(Boolean b);
 
     @POST
-    @Path("/testBody3")
+    @Path("/testBody4")
     @Consumes({MediaType.TEXT_PLAIN})
     TestPO testBody2(TestPO b);
 

--- a/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/extension/resteasy/ResteasyContext.java
+++ b/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/extension/resteasy/ResteasyContext.java
@@ -169,10 +169,13 @@ public interface ResteasyContext {
         for (Map.Entry<String, List<Object>> entry : headers.entrySet()) {
 
             String key = entry.getKey();
-            if (entry.getValue() == null) {
+            List<Object> value = entry.getValue();
+            if (value == null || value.isEmpty()) {
                 continue;
             }
-            response.addOutputHeaders(key, entry.getValue().toString());
+            for (Object tmp : value) {
+                response.addOutputHeaders(key, tmp.toString());
+            }
         }
     }
 

--- a/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/extension/resteasy/filter/ResteasyNettyHttpResponse.java
+++ b/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/extension/resteasy/filter/ResteasyNettyHttpResponse.java
@@ -42,7 +42,14 @@ public class ResteasyNettyHttpResponse implements HttpResponse {
         for (Map.Entry<String, List<String>> headers : outputHeaders.entrySet()) {
             String key = headers.getKey();
             List<String> value = headers.getValue();
-            multivaluedMap.add(key, value);
+
+            if (value == null || value.isEmpty()) {
+                continue;
+            }
+
+            for (String val : value) {
+                multivaluedMap.add(key, val);
+            }
         }
     }
 

--- a/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/message/codec/JsonCodec.java
+++ b/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/message/codec/JsonCodec.java
@@ -33,13 +33,13 @@ import java.util.Set;
 /**
  * body is json
  */
-@Activate("json")
+@Activate(value = "json", order = 100)
 public class JsonCodec implements HttpMessageCodec<byte[], OutputStream> {
     private static final Set<Class> unSupportClasses = new HashSet<>();
 
     static {
         unSupportClasses.add(byte[].class);
-        unSupportClasses.add(String.class);
+        //        unSupportClasses.add(String.class);
     }
 
     public static void addUnSupportClass(Class<?> unSupportClass) {

--- a/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/message/codec/JsonCodec.java
+++ b/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/message/codec/JsonCodec.java
@@ -39,7 +39,6 @@ public class JsonCodec implements HttpMessageCodec<byte[], OutputStream> {
 
     static {
         unSupportClasses.add(byte[].class);
-        //        unSupportClasses.add(String.class);
     }
 
     public static void addUnSupportClass(Class<?> unSupportClass) {

--- a/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/message/codec/StringCodec.java
+++ b/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/message/codec/StringCodec.java
@@ -28,7 +28,7 @@ import java.nio.charset.StandardCharsets;
 /**
  *  body is string
  */
-@Activate("string")
+@Activate(value = "string", order = 200)
 public class StringCodec implements HttpMessageCodec<byte[], OutputStream> {
 
     @Override

--- a/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/netty/NettyHttpResponse.java
+++ b/dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/netty/NettyHttpResponse.java
@@ -190,6 +190,10 @@ public class NettyHttpResponse implements HttpResponse {
             outputHeaders.put(name, values);
         }
 
+        if (values.contains(value)) {
+            return;
+        }
+
         values.add(value);
     }
 

